### PR TITLE
Banner TS Convert

### PR DESCRIPTION
**Pull request by jaketrent**
## Instructions

- fill in title: format of "<package name>: short description"
- label: attach bug or enhancement request label

### What You're Solving

- Any links to #57 related issues
- Explanation of the purpose of the code

### Design Decisions

- Why you made certain design decisions
- Anything to watch out for
- Any input you're seeking related to the design
- Tradeoffs you conciously made

### How to Verify

- Steps needed to independently verify additions

### CODE REVIEW

File: packages/banner/package.json [hunk 1]
```diff
@@ -9,22 +9,20 @@
...
+    "build:cjs": "tsc --project tsconfig.build.json --module commonjs --target es5 --outDir dist/cjs",
     "build:css": "build-css --useGlamor -i dist/cjs/css/index.js",
-    "build:js": "cross-env ESM=true babel --root-mode upward src --out-dir dist/esm",
+    "build:esm": "tsc --project tsconfig.build.json --module es2015 --target es5 --outDir dist/esm",
     "build:watch": "yarn build:js -- --watch",
     "storybook": "start-storybook -p 6006",
-    "test": "jest",
+    "test": "tsc --noEmit --project ./tsconfig.json && jest",
     "test:watch": "yarn test -- --watchAll"
   },
```
Comment (danethurber) [inline packages/banner/package.json:20]: add typing and style reference please

File: packages/banner/src/react/__specs__/storyshots.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import path from 'path'
```
Comment (danethurber) [inline packages/banner/src/react/__specs__/storyshots.spec.tsx:1]: noticed this was a `tsx` file. i thought we were doing `ts` extensions for this file but i'm not really sure 😄

**Review by danethurber — CHANGES_REQUESTED** [review]
- use `keyMirror` in vars
- missing the `tsconfig.build.json` file
- `strict` mode?

File: packages/banner/src/react/__specs__/storyshots.spec.tsx (hunk 2 above)
Comment (jaketrent) [reply packages/banner/src/react/__specs__/storyshots.spec.tsx:1]: Yeah, in this case, looks like it doesn't need to be tsx.  Some have mocks in them with jsx, those do. Anyway, doesn't apply here.

On the other stories, in the past I:

1. Had an error on createMockNode line
2. Change file extension, and it worked
3. Didn't do strict mode

Now that we're doing strict mode:

1. Had error on createMockNode line
2. Removed that line and the strict tsc error goes away
3. Don't have to redo file ext.

🤷 😆

**Comment by jaketrent** [comment]
Made changes. Ready for re-review.

File: packages/banner/jest.config.js [hunk 3]
```diff
@@ -5,5 +5,5 @@ module.exports = {
   ...baseConfig,
   displayName: name,
   name: name,
-  testMatch: [`${__dirname}/**/*/?(*.)+(spec|test).js`]
+  testMatch: [`${__dirname}/**/*/?(*.)+(spec|test).(js|ts|tsx)`]
```
Comment (EdwardIrby) [inline packages/banner/jest.config.js:8]: Are there still js spec files being tested?

File: packages/banner/package.json [hunk 4]
```diff
@@ -8,23 +8,23 @@
...
-    "build:cjs": "babel --root-mode upward src --out-dir dist/cjs",
+    "build": "run-s build:cjs build:esm build:css",
+    "build:cjs": "tsc --project tsconfig.build.json --module commonjs --target es5 --outDir dist/cjs",
     "build:css": "build-css --useGlamor -i dist/cjs/css/index.js",
-    "build:js": "cross-env ESM=true babel --root-mode upward src --out-dir dist/esm",
+    "build:esm": "tsc --project tsconfig.build.json --module es2015 --target es5 --outDir dist/esm",
     "build:watch": "yarn build:js -- --watch",
     "storybook": "start-storybook -p 6006",
-    "test": "jest",
+    "test": "tsc --noEmit --project ./tsconfig.json && jest",
```
Comment (EdwardIrby) [inline packages/banner/package.json:20]: Haven't been changing this didn't realize it was necessary?

**Review by EdwardIrby — APPROVED** [review]

File: packages/banner/package.json (hunk 4 above)
Comment (jaketrent) [reply packages/banner/package.json:20]: I don't know that it's necessary, but I think the tsc step in test seems like a good precaution.

File: packages/banner/jest.config.js (hunk 3 above)
Comment (jaketrent) [reply packages/banner/jest.config.js:8]: there aren't .js files, no.  We could remove that. 🤷

**Pull request merged**